tool/lsm.go

```diff
@@ -288,6 +288,9 @@ func (l *lsmT) buildEdits(edits []*manifest.VersionEdit) error {
 
 		for j := range ve.NewFiles {
 			nf := &ve.NewFiles[j]
+			if nf.Meta.Virtual {
+				nf.Meta.InitProviderBacking(nf.BackingFileNum)
```

Comment:
I think what you want to do is to have a running map of BackingFileNums -> FileBackings, populated by `ve.CreatedBackingTables`, and then set `nf.Meta.FileBacking = backingsMap[nf.BackingFileNum]` here.

It's possible the backing is a non-provider backing, eg. virtual sstables backed by a local sstable (created due to an ingest split), in which case this won't work.

Reply:
done